### ISSUE TITLE
`AbstractClassCanBeInterface` false positive for sealed classes that inherit another class

### ISSUE THREAD

**Issue by UnguidedFreedom** [detekt/detekt#9182]
## Expected Behavior
No `AbstractClassCanBeInterface` for sealed classes without members that inherit another class as it cannot be an interface.

## Observed Behavior
The rule is triggered.

## Steps to Reproduce
```kotlin
sealed class MyError : Throwable() {
    data object FirstType : MyError()
    data object SecondType : MyError()
}
```

## Context
Noticed this new violation when trialing out the latest snapshot, forcing me to put a `@Suppress`

## Your Environment

* Version of detekt used: `main-SNAPSHOT`
* Version of Gradle used (if applicable): 9.4.1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: macOS 26.3
* Link to your project (if it's a public repository): N/A
